src/main/java/com/google/starfish/servlets/UploadNotesServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/upload-notes")
+public class UploadNotesServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String blobKey = getUploadedFileBlobKey(request, "notes-pdf");
+    // TODO: Put blobKey and other course note information into database
+    response.sendRedirect("/");
+  }
+
+  /** Returns a blob key for the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileBlobKey(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
```

Comment:
Not sure if you can create a BlobstoreService object outside a method, but if possible, might be good practice to create it as a private field of the Servlet so if there were other methods that used it then they wouldn't have to make an instance of their own BlobstoreService.

Reply:
Yep you can, I'll do that!